### DISCUSSION TITLE
Is it possible to avoid creating logger three times?

### DISCUSSION THREAD

**Discussion by michaelplavnik** [VIP-Logic/NBomber#735] [General]
The problem is that some sinks need to be properly configured/update to work in such way.

Do I understand correctly, that check for prior call to CreateLogger() was caused by need for Enrichment?

What will happen if try catch is removed?

```    
let withLoggerConfig (createLoggerConfig: unit -> LoggerConfiguration) (context: NBomberContext) =
        try
            // this is limitation of Serilog
            // to invoke CreateLogger() twice on the same instance of LoggerConfiguration
            // it's why we can't just accept LoggerConfiguration
            createLoggerConfig().CreateLogger() |> ignore
            createLoggerConfig().CreateLogger() |> ignore
        with
        | :? InvalidOperationException ->
            failwith "createLoggerConfig should always return a new instance of LoggerConfiguration"

        { context with CreateLoggerConfig = Some createLoggerConfig }
```

**Comment by AntyaDev** [comment]
Hi @michaelplavnik 
Yes, this check is essential, especially for Cluster mode. For some reason, Serilog doesn't allow it to reset the **Enrichment** properties that are attached to each message. 

@michaelplavnik  can you please elaborate about your particular case with Sink?

**Comment by michaelplavnik** [comment]
Easy example is file logger. Without setting shared or rolling file option, it logs custom events, but not NBomber events.

I do not completely understand relation of logging with Cluster mode. That said following might provide alternative view
- Centralized logging (for instance Seq) is readily available
- Redirecting log to coordinator looks like out-of-NBomber-box assembly.
- Consider replacing setting enrichment options hidden in NBomber runtime with custom enricher. If developer wants enriched logging, it will configure it. Custom enricher can use AsyncLocal to access current NBomber context.

**Reply by AntyaDev** [reply]
> Easy example is file logger. Without setting shared or rolling file option, it logs custom events, but not NBomber events.

Can you show some code examples? I would like to reproduce.
